app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -55,8 +55,7 @@ def auto_attach
     api :GET, "/hosts/:host_id/subscriptions/events", N_("List subscription events for the host")
     param :host_id, Integer, :desc => N_("Id of the host"), :required => true
     def events
-      collection = full_result_response(@host.subscription_facet.candlepin_consumer.events.map { |e| OpenStruct.new(e) })
-      respond_for_index :collection => collection
+      respond_for_index :collection => {}
```

Comment:
I think this should technically be an Array rather than Hash

Reply:
@jturel collection was a hash, that's why I had returned an empty hash earlier. I am getting this error message if I use an array.
`{"displayMessage":"no implicit conversion of Symbol into Integer","errors":["no implicit conversion of Symbol into Integer"]}`

Ideally the events api should be returning the below - 
`{"total":null,"subtotal":null,"page":null,"per_page":null,"error":null,"search":null,"sort":{"by":null,"order":null},"results":null}`

So I'm changing it back to an empty hash. Let me know if that makes sense.

Reply:
I see what's happening. The collection should actually be a hash as you mentioned because of the view template:


https://github.com/Katello/katello/blob/32e7c74eba444e9c20048b0e587b1aca2d19cf3e/app/views/katello/api/v2/host_subscriptions/events.json.rabl#L5


I think `results` should not be null in the response, though. That could break scripts or maybe even hammer in some way (not sure) if someone is trying to iterate over null. Can you put a `results` key in the hash with value of []?

Reply:
Okay yes. That makes sense.